src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.cpp

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "stdafx.h"
+
+#if defined(INCLUDE_AZUREKINECT)
+#include "AzureKinectFrameProvider.h"
+#include <k4a/k4a.h>
+#if defined(INCLUDE_AZUREKINECT_BODYTRACKING)
+#include <k4abt.h>
+#endif
+
+AzureKinectFrameProvider::AzureKinectFrameProvider(ProviderType providerType)
+    : detectMarkers(false)
+    , markerSize(0.0f)
+    , _captureFrameIndex(0)
+    , markerDetector(new ArUcoMarkerDetector())
+    , _colorSRV(nullptr)
+    , _depthSRV(nullptr)
+    , _bodySRV(nullptr)
+    , calibration()
+    , d3d11Device(nullptr)
+    , k4aDevice(nullptr)
+    , lock()
+    , transformation(nullptr)
+    , transformedDepthImage(nullptr)
+    , transformedBodyMaskImage(nullptr)
+    , bodyMaskImage(nullptr)
+{
+    switch (providerType)
+    {
+        case AzureKinect_DepthCamera_Off:
+            depthCameraMode = K4A_DEPTH_MODE_OFF;
+            break;
+        case AzureKinect_DepthCamera_NFOV:
+            depthCameraMode = K4A_DEPTH_MODE_NFOV_UNBINNED;
+            break;
+        case AzureKinect_DepthCamera_WFOV:
+            depthCameraMode = K4A_DEPTH_MODE_WFOV_2X2BINNED;
+            break;
+        default:
+            depthCameraMode = K4A_DEPTH_MODE_OFF;
+            break;
+    }
+}
+
+HRESULT AzureKinectFrameProvider::Initialize(ID3D11ShaderResourceView* colorSRV, ID3D11ShaderResourceView* depthSRV, ID3D11ShaderResourceView* bodySRV, ID3D11Texture2D* outputTexture)
+{
+    InitializeCriticalSection(&lock);
+
+    _captureFrameIndex = 0;
+    _depthSRV = depthSRV;
+    _bodySRV = bodySRV;
+    _colorSRV = colorSRV;
+    _colorSRV->GetDevice(&d3d11Device);
+
+    if (K4A_RESULT_SUCCEEDED != k4a_device_open(K4A_DEVICE_DEFAULT, &k4aDevice))
+    {
+        OutputDebugString(L"Failed to open AzureKinect device");
+        goto FailedExit;
+    }
+
+    config.color_format = K4A_IMAGE_FORMAT_COLOR_BGRA32;
+    config.color_resolution = K4A_COLOR_RESOLUTION_1080P;
+    config.depth_mode = depthCameraMode;
+    config.camera_fps = K4A_FRAMES_PER_SECOND_30;
```

Comment:
>K4A_FRAMES_PER_SECOND_30 [](start = 24, length = 24)

We may want to keep this at 1080p but would we want this 30fps to be configurable?

Reply:
30 fps is the maximum supported, and I don't think we want to go lower than that

---
In reply to: [384179632](https://github.com/microsoft/MixedReality-SpectatorView/pull/327#discussion_r384179632) [](ancestors = 384179632)